vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,88 @@ For more canonical documentation of defining packages dependency check the offic
 
 Some of internally used C routines are now exported on C level thus can be used in R packages directly from their C code. See [`?cdt`](https://rdatatable.gitlab.io/data.table/reference/cdt.html) for details and [Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html) _Linking to native routines in other packages_ section for usage.
 
-## Importing from non-R Applications {non-r-API}
+## Importing from non-r Applications {non-r-api}
 
 Some tiny parts of `data.table` C code were isolated from the R C API and can now be used from non-R applications by linking to .so / .dll files. More concrete details about this will be provided later; for now you can study the C code that was isolated from the R C API in [src/fread.c](https://github.com/Rdatatable/data.table/blob/master/src/fread.c) and [src/fwrite.c](https://github.com/Rdatatable/data.table/blob/master/src/fwrite.c).
+
+## How to convert your Depends dependency on data.table to Imports
+
+To convert a `Depends` dependency on `data.table` to an `Imports` dependency in an R package, follow these steps:
+
+### 1. Update the DESCRIPTION File
+In your package's `DESCRIPTION` file, change the entry from `Depends` to `Imports`. This involves moving `data.table` from the `Depends` field to the `Imports` field.
+
+**Before:**
+```r
+Depends:
+    R (>= 3.5.0),
+    data.table
+Imports:
+```
+
+**After:**
+```r
+Depends:
+    R (>= 3.5.0)
+Imports:
+    data.table
+```
+
+### 2. Modify the NAMESPACE File
+Ensure that any functions from `data.table` that you use in your package are imported. This can be done either by explicitly importing specific functions or by importing the entire package but specific imports are considered best practice.
+
+#### Importing Specific Functions
+If you use specific functions like `data.table()`, `setDT()`, or `merge()`, you can import them individually in the `NAMESPACE` file:
+
+```r
+importFrom(data.table, data.table, setDT, merge)
+```
+
```

Comment:
Note that this is not complete. Users will _also_ need to make sure they `importFrom(data.table, S3Generics)` for any S3 generic owned by data.table for which they are adding a method in their package.

Readers should check our `NAMESPACE` file for the latest set of S3 generics (is there a way to get this dynamically? I think so but forget how off the top of my head), e.g. `as.data.table()`, `groupingsets()`, `as.IDate()`, etc.

Reply:
More generally, they'd need to do other stuff for S4 methods, which they might need since we do `exportClasses()`:

https://github.com/Rdatatable/data.table/blob/6a4bd543a62cc4cee7b9bc20cb8afbcb13732091/NAMESPACE#L5

But that should be rare -- I don't see anyone doing that specifically on CRAN. I _do_ see some people importing `data.table` as a class, which I'm not quite sure I understand:

https://github.com/search?q=org%3Acran%20%2FimportClassesFrom%5B(%5Ddata%5B.%5Dtable%2F&type=code